joint_state_broadcaster/src/joint_state_broadcaster.cpp

```diff
@@ -229,6 +229,7 @@ bool has_any_key(
 
 bool JointStateBroadcaster::init_joint_data()
 {
+  joint_names_.clear()
```

Comment:
This method is called in `on_activate` to make this even more correct, I propose to add this fix into `on_deactivate` method. This is somehow more in line with the controller's lifecycle. @bmagyar what do you think*

Reply:
That's technically be more correct but I think it's even more bulletproof to let the function generating a new setup do a quick cleanup -- as-is now